distributed/utils.py

```diff
@@ -1489,3 +1489,29 @@ def format_dashboard_link(host, port):
     else:
         scheme = "http"
     return template.format(scheme=scheme, host=host, port=port, **os.environ)
+
+
+class Log(str):
+    """A container for logs."""
+
+    def _widget(self):
+        from ipywidgets import HTML
+
+        return HTML(value="<pre><code>{logs}</code></pre>".format(logs=self))
+
+    def _ipython_display_(self, **kwargs):
+        return self._widget()._ipython_display_(**kwargs)
+
+
+class Logs(dict):
+    """A container for multiple logs."""
+
+    def _widget(self):
+        from ipywidgets import Accordion
+
+        accordion = Accordion(children=[log._widget() for log in self.values()])
+        [accordion.set_title(i, title) for i, title in enumerate(self.keys())]
+        return accordion
```

Comment:
Thoughts on using details and summary tags instead here?

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

I think that it's nicer to use raw HTML and `_repr_html_` over widgets if possible, both for simplicity, and for static notebook rendering.

Reply:
That would definitely be neater. However there is [no support for those tags in Microsoft flavor browsers](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#Browser_compatibility).

Is that an acceptable trade off?

Reply:
Hrm, that's a shame. 

> Is that an acceptable trade off?

I don't know.  I suspect that you know more about the deployment of browsers and eventual adoption of these tags.

Reply:
I expect usage of Microsoft browsers within the Dask community is low. However that is just a feeling and not backed up with any evidence, might be nice to try and find out more specifically (shame we missed the survey). If we look at [stats for the general population](https://en.wikipedia.org/wiki/Usage_share_of_web_browsers) usage of these browsers is around 15%.

Microsoft have a track record of refusing to implement some standards in their browsers. Given that all other browsers support these tags it's likely that they do not intend to implement these.

I feel like introducing something that will not work for up to 15% of users isn't good. If that number was 5% of lower I would be fine with it.

Reply:
What happens in MS browsers?  Do they fall back sensibly?  Perhaps they just render the summaries?  If so then this might not be that bad.

Reply:
I was curious enough to fire up my VM.  The details/summary tags seem to work, at least on IE11, even if they aren't officially supported.

Reply:
Ah, no, details/summary don't work, but do they do seem to just render the fully expanded version, which seems good enough for me.

Reply:
> Ah, no, details/summary don't work, but do they do seem to just render the fully expanded version, which seems good enough for me.